pkg/controllers/metrics/node/controller.go

```diff
@@ -0,0 +1,340 @@
+	metricLabels[nodeCapacityType] = node.Labels[v1alpha5.LabelCapacityType]
+	metricLabels[nodeInstanceType] = node.Labels[v1.LabelInstanceTypeStable]
+	metricLabels[nodePhase] = string(node.Status.Phase)
+	// Add node labels
+	labels, err := json.Marshal(node.GetLabels())
+	if err != nil {
+		return nil, fmt.Errorf("marshal pod labels: %w", err)
+	}
+	metricLabels[nodeLabels] = string(labels)
+	return metricLabels, nil
+}
+
+func (c *Controller) updateGauges(ctx context.Context, node *v1.Node) error {
+
+	// Calculate total requests and limits from all pods
+	podlist := &v1.PodList{}
+	if err := c.KubeClient.List(ctx, podlist, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
+		return fmt.Errorf("listing pods on node %s, %w", node.Name, err)
+	}
+	reqs, limits := getPodsTotalRequestsAndLimits(podlist.Items)
+	allocatable := node.Status.Capacity
+	overheads := calculateDaemonSetOverhead(podlist.Items)
+	if len(node.Status.Allocatable) > 0 {
+		allocatable = node.Status.Allocatable
+		// calculating system daemons overhead
+		for resourceName, quantity := range allocatable {
+			overhead := node.Status.Capacity[resourceName]
+			overhead.Sub(quantity)
+			overheads[resourceName] = overhead
+		}
+	}
+
+	// Populate overhead metrics
+	if err := c.insertGaugeValues(overheads, node, overheadGaugeVec); err != nil {
+		logging.FromContext(ctx).Debugf("Failed to generate gauge: %w", err)
+	}
+	// Populate total request metrics
+	if err := c.insertGaugeValues(reqs, node, requestsGaugeVec); err != nil {
+		logging.FromContext(ctx).Debugf("Failed to generate gauge: %w", err)
+	}
+	// Populate total limits metrics
+	if err := c.insertGaugeValues(limits, node, limitsGaugeVec); err != nil {
+		logging.FromContext(ctx).Debugf("Failed to generate gauge: %w", err)
+	}
+	// Populate allocatable metrics
+	if err := c.insertGaugeValues(allocatable, node, allocatableGaugeVec); err != nil {
+		logging.FromContext(ctx).Debugf("Failed to generate gauge: %w", err)
+	}
+
+	return nil
+}
+
+func (c *Controller) insertGaugeValues(resources map[v1.ResourceName]resource.Quantity, node *v1.Node, gaugeVec *prometheus.GaugeVec) error {
+
+	for resourceName, quantity := range resources {
+		resourceTypeName := strings.ReplaceAll(strings.ToLower(string(resourceName)), "-", "_")
+		labels, err := c.generateLabels(node, resourceTypeName)
+		// Register the set of labels that are generated for node
+		nodeNamespacedName := types.NamespacedName{Name: node.Name}
+		c.LabelsMap[nodeNamespacedName] = append(c.LabelsMap[nodeNamespacedName], labels)
+		if err != nil {
+			return fmt.Errorf("generate new labels: %w", err)
+		}
+		gauge, err := gaugeVec.GetMetricWith(labels)
+		if err != nil {
+			return fmt.Errorf("generate new gauge: %w", err)
+		}
+		if resourceName == v1.ResourceCPU {
+			gauge.Set(float64(quantity.MilliValue()))
+		} else {
+			gauge.Set(float64(quantity.Value()))
+		}
+	}
+	return nil
+}
+
+func calculateDaemonSetOverhead(pods []v1.Pod) v1.ResourceList {
```

Comment:
This misses out on kube reserved 
![image](https://user-images.githubusercontent.com/2585992/146578704-bdee3f0b-efb5-4310-a797-683b9af3bbf4.png)

Reply:
Yes now refactored to return system overhead (the three gray boxes) and daemonset pod separatly.